R/clean_names.R

```diff
@@ -68,19 +68,22 @@ clean_names <- function(dat, ...) {
   UseMethod("clean_names")
 }
 
-#' @rdname clean_names
-#' @export
-clean_names.data.frame <- function(dat, ...) {
-  stats::setNames(dat, make_clean_names(names(dat), ...))
-}
-
 #' @rdname clean_names
 #' @export
 clean_names.default <- function(dat, ...) {
-  stop(
-    "No `clean_names()` method exists for the class ", paste(class(dat), collapse=", "),
-    "\nConsider janitor::make_clean_names() for other cases of manipulating vectors of names."
-  )
+  if(is.null(names(dat)) && is.null(dimnames(dat))) {
+    stop(
+      "`clean_names()` requires that either names or dimnames be non-null.",
+      call. = FALSE
+    )
+    return(dat)
```

Comment:
Could you add a test for this line?  Codecov is reporting a failure b/c this line is untested.  Though is it unreachable due to the preceding `stop()`?  If so it could just be removed.

Reply:
The `return(dat)` was a leftover from the time this stop was a warning. I removed it